services/director-v2/src/simcore_service_director_v2/api/routes/dynamic_services.py

```diff
@@ -1,20 +1,225 @@
+from fastapi import APIRouter, Depends, Header
+from fastapi.responses import RedirectResponse
+from models_library.projects import ProjectID
+from models_library.service_settings import SimcoreService
+from models_library.services import ServiceKeyVersion
+from simcore_service_director_v2.models.schemas.constants import UserID
 from starlette import status
 from starlette.datastructures import URL
 
-from ...models.domains.dynamic_services import RetrieveDataIn, RetrieveDataOutEnveloped
+from ...core.settings import DynamicServicesSettings
+from ...models.domains.dynamic_services import (
+    DynamicServiceCreate,
+    DynamicServiceOut,
+    RetrieveDataIn,
+    RetrieveDataOutEnveloped,
+)
+from ...models.schemas.constants import (
+    DYNAMIC_PROXY_SERVICE_PREFIX,
+    DYNAMIC_SIDECAR_SERVICE_PREFIX,
+    UserID,
+)
+from ...modules.dynamic_sidecar.docker_utils import (
+    is_dynamic_service_running,
+    list_dynamic_sidecar_services,
+)
+from ...modules.dynamic_sidecar.exceptions import DynamicSidecarNotFoundError
+from ...modules.dynamic_sidecar.monitor import DynamicSidecarsMonitor, get_monitor
+from ...modules.dynamic_sidecar.monitor.models import MonitorData
+from ...modules.dynamic_sidecar.service_specs import assemble_service_name
 from ...utils.logging_utils import log_decorator
+from ..dependencies.director_v0 import DirectorV0Client, get_director_v0_client
 from ..dependencies.dynamic_services import (
     ServicesClient,
     get_service_base_url,
     get_services_client,
+    get_settings,
 )
 
 router = APIRouter()
-logger = logging.getLogger(__file__)
+log = logging.getLogger(__file__)
+
+TEMPORARY_PORT_NUMBER = 65_534
+
+
+@router.get(
+    "",
+    status_code=status.HTTP_200_OK,
+    response_model=List[DynamicServiceOut],
+    summary=(
+        "returns a list of running interactive services filtered by user_id and/or project_id"
+        "both legacy (director-v0) and modern (director-v2)"
+    ),
+)
+async def list_running_dynamic_services(
+    user_id: Optional[UserID] = None,
+    project_id: Optional[ProjectID] = None,
+    director_v0_client: DirectorV0Client = Depends(get_director_v0_client),
+    dynamic_services_settings: DynamicServicesSettings = Depends(get_settings),
+    monitor: DynamicSidecarsMonitor = Depends(get_monitor),
+) -> List[Dict[str, str]]:
+    legacy_running_services: List[
+        DynamicServiceOut
+    ] = await director_v0_client.get_running_services(user_id, project_id)
+
+    get_stack_statuse_tasks: List[DynamicServiceOut] = [
+        monitor.get_stack_status(service["Spec"]["Labels"]["uuid"])
+        for service in await list_dynamic_sidecar_services(
+            dynamic_services_settings.dynamic_sidecar, user_id, project_id
+        )
+    ]
+    modern_running_services: List[Dict[str, str]] = [
+        x.dict(exclude_unset=True)
+        for x in await asyncio.gather(*get_stack_statuse_tasks)
+    ]
+
+    return legacy_running_services + modern_running_services
```

Comment:
you're adding a List[DynamicServiceOut] with a List[Dict[str,Any]]. there is something fishy there.
the response_model says it returns a List[DynamicServiceOut]. please adjust, there is no need to convert to a dict, FastApi takes care of that.

Reply:
refactored the function and return types.

Reply:
Think I found out how to make both happy (`response_model_exclude_unset=True`); will refactor.